src/WebJobs.Script.WebHost/ProxyFunctionExecutor.cs

```diff
@@ -22,19 +22,87 @@ public class ProxyFunctionExecutor : IFuncExecutor
         {
             _scriptHostManager = scriptHostManager;
             _webHookReceiverManager = webHookReceiverManager;
             _secretManager = secretManager;
+
+            _httpPrefix = "api/";
+
+            if (_scriptHostManager.Instance != null)
+            {
+                var json = _scriptHostManager.Instance.ScriptConfig.HostConfig.HostConfigMetadata;
+
+                if (json != null && json["http"] != null && json["http"]["routePrefix"] != null)
+                {
+                    _httpPrefix = json["http"]["routePrefix"].ToString() + "/";
+                }
+            }
         }
 
         public async Task ExecuteFuncAsync(string funcName, Dictionary<string, object> arguments, CancellationToken cancellationToken)
         {
             HttpRequestMessage request = arguments[ScriptConstants.AzureFunctionsHttpRequestKey] as HttpRequestMessage;
-            var function = _scriptHostManager.GetHttpFunctionOrNull(request);
+
+            FunctionDescriptor function = null;
+            var path = request.RequestUri.AbsolutePath.Trim(new char[] { '/' });
+
+            if (path.StartsWith(_httpPrefix))
+            {
+                path = path.Remove(0, _httpPrefix.Length);
+            }
+
+            // This is a call to the local function app, before handing the route to asp.net to pick the FunctionDescriptor the following will be run:
+            // 1. If the request maps to a local http trigger function name then that function will be picked.
+            // 2. Else if the request maps to a custom route of a local http trigger function then that function will be picked
+            // 3. Otherwise the request will be given to asp.net to pick the appropriate route.
+            foreach (var func in _scriptHostManager.HttpFunctions.Values)
+            {
+                if (!func.Metadata.IsProxy)
+                {
+                    if (path.Equals(func.Metadata.Name, StringComparison.OrdinalIgnoreCase))
+                    {
+                        function = func;
+                        break;
+                    }
+                    else
+                    {
+                        foreach (var binding in func.InputBindings)
+                        {
+                            if (binding.Metadata.IsTrigger)
+                            {
+                                string functionRoute = null;
+                                var jsonContent = binding.Metadata.Raw;
+                                if (jsonContent != null && jsonContent["route"] != null)
+                                {
+                                    functionRoute = jsonContent["route"].ToString();
+                                }
+
+                                if (!string.IsNullOrEmpty(functionRoute) && path.Equals(functionRoute, StringComparison.OrdinalIgnoreCase))
```

Comment:
Routes can contain regular expressions and other constraints. How can a string comparison like this work?

Reply:
Yes this only work for static routes. I can open a separate issue for dynamic routes and investigate later.

---
In reply to: [146596316](https://github.com/Azure/azure-webjobs-sdk-script/pull/2060#discussion_r146596316) [](ancestors = 146596316)